drivers/net/mlx5/mlx5_vdpa.c

```diff
@@ -360,6 +520,155 @@ mlx5_vdpa_setup_notify_relay(struct vdpa_priv *priv)
 	}
 	return 0;
 }
+static int
+mlx5_vdpa_dma_map(struct vdpa_priv *priv)
+{
+	uint32_t i;
+	int ret;
+	struct rte_vhost_memory *mem = NULL;
+	struct mlx5_devx_mkey_attr mkey_attr;
+	static int klm_index;
+	struct mlx5_vdpa_query_mr *entry = NULL;
+	struct mlx5_vdpa_query_mr_list *list_elem = NULL;
+	struct rte_vhost_mem_region *reg = NULL;
+	/*TODO(liel): Working with KLM size of 1GB, change to GCD on empty_mr and region sizes*/
+	uint64_t klm_size = 1073741824;
+	uint64_t min_size = klm_size;
+	uint64_t mem_size;
+
+	ret = rte_vhost_get_mem_table(priv->id, &mem);
+	if (ret < 0) {
+		DRV_LOG(ERR, "failed to get VM memory layout.");
+		return -1;
+	}
+	for (i = 0; i < (mem->nregions); i++) {
+		if (min_size > mem->regions[i].size) {
+			DRV_LOG(ERR, "Min region size is smaller than KLM size.");
+			return -1;
+		}
```

Comment:
@bekets 

Sorry i didn't had time to review before you merged.

haven't reviewed it all, only this function.

one important issue is the assumption the current code make on the size of stride (1G). 
it is not enough to check the min size rather also need to check the empty region size. 
or the create mkey will fail if the klm sizes are not equal?